src/ucp/tag/tag_match.inl

```diff
@@ -99,6 +99,17 @@ ucp_tag_exp_add(ucp_tag_match_t *tm, ucp_request_t *req)
     ucp_tag_exp_push(tm, ucp_tag_exp_get_req_queue(tm, req), req);
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_tag_exp_delete(ucp_request_t *req, ucp_tag_match_t *tm,
+                   ucs_queue_head_t *queue, ucs_queue_iter_t iter)
+{
+    if (req->flags & UCP_REQUEST_FLAG_BLOCK_OFFLOAD) {
```

Comment:
can we know that when getting SW eager message, this condition will always be true?

Reply:
the same flow is used for unexpected tag offload messages

Reply:
but maybe we can pass some flag which would be resolved in compile time to avoid the check for sw unexpected?

Reply:
instead of passing 'req' to this function, for example

Reply:
I can't distinguish whether the message arrived unexpectedly in HW or it was not posted at all.